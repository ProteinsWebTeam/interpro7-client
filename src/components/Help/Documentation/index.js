// @flow
import React, { PureComponent } from 'react';

import Link from 'components/generic/Link';
import loadable from 'higherOrder/loadable';
import { schemaProcessDataPageSection } from 'schema_org/processors';

import { foundationPartial } from 'styles/foundation';

import fonts from 'EBI-Icon-fonts/fonts.css';
import ipro from 'styles/interpro-new.css';
import ebiGlobalStyles from 'ebi-framework/css/ebi-global.css';
import local from './style.css';

import { InterProLogo } from 'components/Header/Title';

const f = foundationPartial(ebiGlobalStyles, fonts, ipro, local);

const SchemaOrgData = loadable({
  loader: () => import(/* webpackChunkName: "schemaOrg" */ 'schema_org'),
  loading: () => null,
});
export default class Documentation extends PureComponent /*:: <{}> */ {
  render() {
    return (
      <section>
        <SchemaOrgData
          data={{
            name: 'InterPro Documentation Page',
            description:
              'Includes links to the documentations for the different parts of our service',
          }}
          processData={schemaProcessDataPageSection}
        />
        <div className={f('row')}>
          <div className={f('columns', 'large-8')}>
            <div className={f('flex-container')}>
              <div className={f('flex-box')}>
                <InterProLogo /> <h5>Interpro</h5>
                <li>
                  <ul>
                    <li>
                      <Link
                        to={{
                          description: {
                            other: ['about', 'interpro'],
                          },
                        }}
                      >
                        About InterPro
                      </Link>
                    </li>
                    <li>
                      <Link
                        to={{
                          description: {
                            other: ['release_notes'],
                          },
                        }}
                      >
                        Release Notes
                      </Link>
                    </li>
                    <li>
                      <Link
                        to={{
                          description: {
                            other: ['help', 'faqs'],
                          },
                        }}
                      >
                        Frequently Asked Questions (FAQs)
                      </Link>
                    </li>
                    <li>
                      <Link
                        to={{
                          description: {
                            other: ['about', 'ida'],
                          },
                        }}
                      >
                        InterPro Domain Architecture
                      </Link>
                    </li>
                  </ul>
                </li>
              </div>
              <div className={f('flex-box')}>
                <span
                  className={f('icon', 'icon-common', 'xl')}
                  data-icon="&#xf2fd;"
                />{' '}
                <h5>InterProScan</h5>
                <ul>
                  <li>
                    <Link
                      to={{
                        description: {
                          other: ['about', 'interproscan'],
                        },
                      }}
                    >
                      About InterProScan
                    </Link>
                  </li>
                  <li>
                    <Link
                      href="//github.com/ebi-pf-team/interproscan/wiki"
                      className={f('ext')}
                      target="_blank"
                    >
                      <span
                        className={f('icon', 'icon-common')}
                        data-icon="&#xf09b;"
                      />{' '}
                      InterProScan Wiki
                    </Link>
                  </li>
                </ul>
              </div>
              <div className={f('flex-box')}>
                <span
                  className={f('icon', 'icon-common', 'xl')}
                  data-icon="&#xf085;"
                />
                <h5>Technical specs</h5>
                <ul>
                  <li>
                    <Link
                      href="https://github.com/ProteinsWebTeam/interpro7-api/tree/master/docs"
                      className={f('ext')}
                      target="_blank"
                    >
                      <span
                        className={f('icon', 'icon-common')}
                        data-icon="&#xf09b;"
                      />{' '}
                      General API documentation
                    </Link>
                  </li>
                  <li>
                    <Link
                      href="//www.ebi.ac.uk/interpro/api/static_files/swagger/"
                      className={f('ext')}
                      target="_blank"
                    >
                      Swagger API documentation
                    </Link>
                  </li>
                  <li>
                    <Link
                      to={{
                        description: {
                          main: { key: 'result' },
                          result: { type: 'download' },
                        },
                        hash: '/entry/interpro',
                      }}
                    >
                      Code snippet generator
                    </Link>
                  </li>
                  <li>
                    <Link
                      href="//www.ebi.ac.uk/Tools/webservices/"
                      className={f('ext')}
                      target="_blank"
                    >
                      Web services
                    </Link>
                  </li>
                  <li>
                    <Link
                      href="//www.ebi.ac.uk/data-protection/privacy-notice/embl-ebi-public-website"
                      className={f('ext')}
                      target="_blank"
                    >
                      Privacy and security
                    </Link>
                  </li>

                  <li>
                    <Link
                      href="//www.ebi.ac.uk/ebisearch/documentation.ebi"
                      className={f('ext')}
                      target="_blank"
                    >
                      Search documentation (EBI)
                    </Link>
                  </li>
                </ul>
              </div>
              <div className={f('flex-box')}>
                <span
                  className={f('icon', 'icon-common', 'xl')}
                  data-icon="&#xf02d;"
                />{' '}
                <h5>Publications</h5>
                <ul>
                  <li>
                    <Link
                      to={{
                        description: {
                          other: ['help', 'documentation', 'publication'],
                        },
                      }}
                    >
                      Show all previous InterPro publications
                    </Link>
                  </li>
                </ul>
              </div>
            </div>
          </div>
          <div className={f('columns', 'large-4')}>
            <div className={f('box-add', 'margin-top-xxlarge')}>
              <h3 className={f('light')}>Contact us/Additional help</h3>
              <ul>
                {
                  // <li>
                  //   <Link href="//www.ebi.ac.uk/support/interpro-general-query">
                  //     <span
                  //       className={f('icon', 'icon-common')}
                  //       data-icon="&#x27a;"
                  //     />{' '}
                  //     Chat with an assistant{' '}
                  //   </Link>
                  // </li>
                }
                <li>
                  <Link
                    href="//www.ebi.ac.uk/support/interpro"
                    target="_blank"
                    withReferrer
                  >
                    <span
                      className={f('icon', 'icon-common')}
                      data-icon="&#x6e;"
                    />{' '}
                    Submit a ticket
                  </Link>
                </li>
              </ul>
            </div>
          </div>
        </div>
      </section>
    );
  }
}
