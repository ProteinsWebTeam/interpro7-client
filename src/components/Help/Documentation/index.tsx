// @flow
import React from 'react';

import config from 'config';
import Link from 'components/generic/Link';
import loadable from 'higherOrder/loadable';
import { schemaProcessDataPageSection } from 'schema_org/processors';

import fonts from 'EBI-Icon-fonts/fonts.css';
import local from './style.css';

import cssBinder from 'styles/cssBinder';
const f = cssBinder(fonts, local);

const SchemaOrgData = loadable({
  loader: () => import(/* webpackChunkName: "schemaOrg" */ 'schema_org'),
  loading: () => null,
});

const Documentation = () => {
  return (
    <section>
      <SchemaOrgData
        data={{
          name: 'InterPro Documentation Page',
          description:
            'Includes links to the documentations for the different parts of our service',
        }}
        processData={schemaProcessDataPageSection}
      />
      <div className={f('vf-grid', 'vf-grid__col-3')}>
        <div className={f('vf-grid__col--span-2')}>
          <div className={f('flex-container')}>
            <div className={f('flex-box')}>
              <span
                className={f('icon', 'icon-common', 'xl')}
                data-icon="&#xf02d;"
              />{' '}
              <h5>Documentations</h5>
              <li>
                <ul>
                  <li>
                    <Link
                      href={config.root.readthedocs.href}
                      className={f('ext')}
                      target="_blank"
                    >
                      InterPro documentation
                    </Link>
                  </li>
                  <li>
                    <Link
                      href="https://pfam-docs.readthedocs.io/en/latest/"
                      className={f('ext')}
                      target="_blank"
                    >
                      Pfam documentation
                    </Link>
                  </li>
                </ul>
              </li>
            </div>
            <div className={f('flex-box')}>
              <span
                className={f('icon', 'icon-common', 'xl')}
                data-icon="&#xf2fd;"
              />{' '}
              <h5>InterProScan</h5>
              <ul>
                <li>
                  <Link
                    href="https://interproscan-docs.readthedocs.io/"
                    className={f('ext')}
                    target="_blank"
                  >
                    InterProScan documentation
                  </Link>
                </li>
                <li>
                  <Link
                    href="//www.ebi.ac.uk/jdispatcher/docs/"
                    className={f('ext')}
                    target="_blank"
                  >
                    Job Dispatcher documentation
                  </Link>
                </li>
              </ul>
            </div>
            <div className={f('flex-box')}>
              <span
                className={f('icon', 'icon-common', 'xl')}
                data-icon="&#xf085;"
              />
              <h5>Technical specs</h5>
              <ul>
                <li>
                  <Link
                    href="https://github.com/ProteinsWebTeam/interpro7-api/tree/master/docs"
                    className={f('ext')}
                    target="_blank"
                  >
                    <span
                      className={f('icon', 'icon-common')}
                      data-icon="&#xf09b;"
                    />{' '}
                    General API documentation
                  </Link>
                </li>
                <li>
                  <Link
                    href="//www.ebi.ac.uk/interpro/api/static_files/swagger/"
                    className={f('ext')}
                    target="_blank"
                  >
                    Swagger API documentation
                  </Link>
                </li>
                <li>
                  <Link
                    to={{
                      description: {
                        main: { key: 'result' },
                        result: { type: 'download' },
                      },
                      hash: '/entry/interpro/|json',
                    }}
                  >
                    Code snippet generator
                  </Link>
                </li>
                <li>
                  <Link
                    href="//www.ebi.ac.uk/data-protection/privacy-notice/embl-ebi-public-website"
                    className={f('ext')}
                    target="_blank"
                  >
                    Privacy and security
                  </Link>
                </li>
                <li>
                  <Link
                    href="//www.ebi.ac.uk/ebisearch/documentation"
                    className={f('ext')}
                    target="_blank"
                  >
                    EBI Search documentation
                  </Link>
                </li>
              </ul>
            </div>
            <div className={f('flex-box')}>
              <span
                className={f('icon', 'icon-common', 'xl')}
                data-icon="&#xf10d;"
              />{' '}
              <h5>Publications</h5>
              <ul>
                <li>
                  <Link
                    href={`${config.root.readthedocs.href}citing.html`}
                    className={f('ext')}
                    target="_blank"
                  >
                    Citing InterPro
                  </Link>
                </li>
                <li>
                  <Link
                    href="https://pfam-docs.readthedocs.io/en/latest/citing-pfam.html"
                    className={f('ext')}
                    target="_blank"
                  >
                    Citing Pfam
                  </Link>
                </li>
              </ul>
            </div>
          </div>
        </div>
        <div className={f('vf-grid__col--span-1')}>
          <div className={f('box-add', 'margin-top-xxlarge')}>
            <h3 className={f('light')}>Contact us/Additional help</h3>
            <ul>
              {
                // <li>
                //   <Link href="//www.ebi.ac.uk/support/interpro-general-query">
                //     <span
                //       className={f('icon', 'icon-common')}
                //       data-icon="&#x27a;"
                //     />{' '}
                //     Chat with an assistant{' '}
                //   </Link>
                // </li>
              }
              <li>
                <Link
                  href="//www.ebi.ac.uk/support/interpro"
                  target="_blank"
                  withReferrer
                >
                  <span
                    className={f('icon', 'icon-common')}
                    data-icon="&#x6e;"
                  />{' '}
                  Submit a ticket
                </Link>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </section>
  );
};

export default Documentation;
