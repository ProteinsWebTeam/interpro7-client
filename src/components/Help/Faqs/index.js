import React, { PureComponent } from 'react';

import Link from 'components/generic/Link';

import { foundationPartial } from 'styles/foundation';

import loadWebComponent from 'utils/load-web-component';

import fonts from 'EBI-Icon-fonts/fonts.css';
import ipro from 'styles/interpro-new.css';
import local from './style.css';

const f = foundationPartial(ipro, fonts, local);

export default class Faqs extends PureComponent {
  componentDidMount() {
    loadWebComponent(() =>
      import(/* webpackChunkName: "interpro-components" */ 'interpro-components').then(
        m => m.InterproType,
      ),
    ).as('interpro-type');
  }

  render() {
    return (
      <section>
        <div className={f('row')}>
          <div className={f('columns', 'large-8')}>
            <h3 className={f('margin-bottom-xlarge')}>
              Frequently asked questions (FAQs)
            </h3>

            <details className={f('accordion-style')}>
              <summary>
                How can I ensure privacy for my sequence searches?
              </summary>
              <p>
                We adhere to EMBL standards on data privacy. However, if you
                have privacy concerns about submitting sequences for analysis
                via the web, the InterProScan software package can be downloaded
                for local installation from the{' '}
                <Link
                  href="ftp://ftp.ebi.ac.uk/pub/software/unix/iprscan"
                  className={f('ext')}
                  target="_blank"
                >
                  {' '}
                  EBI&apos;s FTP server
                </Link>.
              </p>
            </details>
            <details className={f('accordion-style')}>
              <summary>Can I access InterProScan programmatically?</summary>
              <p>
                InterProScan can be accessed programmatically via Web services
                that allow up to one sequence per request, and up to 25 requests
                in parallel (both{' '}
                <Link
                  href="//www.ebi.ac.uk/seqdb/confluence/pages/viewpage.action?pageId=68165103"
                  className={f('ext')}
                  target="_blank"
                >
                  {' '}
                  SOAP
                </Link>{' '}
                and{' '}
                <Link
                  href="//www.ebi.ac.uk/seqdb/confluence/pages/viewpage.action?pageId=68165098"
                  className={f('ext')}
                  target="_blank"
                >
                  {' '}
                  REST
                </Link>-based services are available).
              </p>
            </details>
            <details className={f('accordion-style')}>
              <summary>How do I interpret my InterProScan results?</summary>
              <p>
                Please see the online tutorials section on the{' '}
                <Link href="/help/tutorial/">Training &amp; tutorials</Link>{' '}
                page.
              </p>
            </details>

            <details className={f('accordion-style')}>
              <summary>
                What are entry types and why are they important?
              </summary>
              <div>
                <p>
                  Each InterPro entry is assigned one of a number of types which
                  tell you what you can infer when a protein matches the entry.
                  The entry types are:
                </p>

                <ul className={f('flex-grid')}>
                  <li>
                    {' '}
                    <interpro-type
                      dimension="4em"
                      type="homologous superfamily"
                      aria-label="Entry type"
                    />{' '}
                    <strong>Homologous Superfamily</strong>: A homologous
                    superfamily is a group of proteins that share a common
                    evolutionary origin, reflected by similarity in their
                    structure. Since superfamily members often display very low
                    similarity at the sequence level, this type of InterPro
                    entry is usually based on a collection of underlying hidden
                    Markov models, rather than a single signature.
                  </li>
                  <li>
                    {' '}
                    <interpro-type
                      dimension="4em"
                      type="Family"
                      aria-label="Entry type"
                    />{' '}
                    <strong>Family</strong>: A protein family is a group of
                    proteins that share a common evolutionary origin reflected
                    by their related functions, similarities in sequence, or
                    similar primary, secondary or tertiary structure. A match to
                    an InterPro entry of this type indicates membership of a
                    protein family.
                  </li>
                  <li>
                    {' '}
                    <interpro-type
                      dimension="4em"
                      type="Domain"
                      aria-label="Entry type"
                    />
                    <strong>Domain</strong>: Domains are distinct functional,
                    structural or sequence units that may exist in a variety of
                    biological contexts. A match to an InterPro entry of this
                    type indicates the presence of a domain.
                  </li>
                  <li>
                    {' '}
                    <interpro-type
                      dimension="4em"
                      type="Repeat"
                      aria-label="Entry type"
                    />
                    <strong>Repeat</strong>: A match to an InterPro entry of
                    this type identifies a short sequence that is typically
                    repeated within a protein.
                  </li>
                  <li>
                    {' '}
                    <interpro-type
                      dimension="4em"
                      type="Site"
                      aria-label="Entry type"
                    />
                    <strong>Site</strong>: A match to an InterPro entry of this
                    type indicates a short sequence that contains one or more
                    conserved residues. The type of sites covered by InterPro
                    are active sites, binding sites, post-translational
                    modification sites and conserved sites.
                  </li>
                  <li>
                    {' '}
                    <interpro-type
                      dimension="4em"
                      type="Unknown"
                      aria-label="Entry type"
                    />
                    <strong>Unintegrated signatures</strong>: In addition to
                    signatures that have been grouped into InterPro entries, you
                    can also find unintegrated signatures that might not yet be
                    curated or might not reach InterPro&apos;s standards for
                    integration. However, they may provide information about a
                    protein.
                  </li>
                </ul>
              </div>
            </details>

            <details className={f('accordion-style')}>
              <summary>What are entry relationships?</summary>
              <div>
                <p>
                  InterPro organises its content into hierarchies, where
                  possible. Entries at the top of these hierarchies describe
                  broad families or domains that share higher level structure
                  and/or function, while those entries at the bottom describe
                  more specific functional subfamilies or structural/functional
                  subclasses of domains.
                </p>
                <p>
                  For example, steroid hormone receptors constitute a family of
                  nuclear receptors responsible for signal transduction mediated
                  by steroid hormones, and can be sub-classified into different
                  groups, including the liver X receptor subfamily. This
                  subfamily consists of nuclear receptors that regulate the
                  metabolism of several important lipids, including oxysterols.
                </p>
              </div>
            </details>

            <details className={f('accordion-style')}>
              <summary>What are overlapping entries?</summary>
              <p>
                On the entry page, the relationship between homologous
                superfamilies and other InterPro entries is calculated by
                analysing the overlap between matched sequence sets. An InterPro
                entry is considered related to a homologous superfamily if its
                sequence matches overlap (i.e., the match positions fall within
                the homologous superfamily boundaries) and either the Jaccard
                index (equivalent) or containment index (parent/child) of the
                matching sequence sets is greater than 0.75.
              </p>
            </details>

            <details className={f('accordion-style')}>
              <summary>
                What do the colours mean in the graphical view of matches to my
                protein?
              </summary>
              {
                // Text need an update
              }
              <p>
                The graphical view of InterPro matches show where the signatures
                that match your protein appear on the sequence. There are two
                ways that these graphical &ldquo;blobs&rdquo; can be coloured.
                If you select &ldquo;Colour by: domain relationship&rdquo;, in
                the left hand menu, the domains that are from the same or
                related InterPro entries will be coloured the same, allowing
                easy visualisation of domains we know to be related.
                Unintegrated signatures will always be grey blobs, family
                signatures will always be shown as white, and sites will always
                be black when this option is selected. If you select
                &ldquo;Colour by: member database&rdquo;, each blob in the
                sequence features section will be coloured according to the
                member database that provides the signature, as shown in this
                diagram. However, the sequence summary view will retain the
                domain relationship colour scheme.
              </p>
            </details>

            <details className={f('accordion-style')}>
              <summary>
                Why are there no e-values associated with InterPro entries?
              </summary>
              <p>
                The signatures contained within InterPro are produced in
                different ways by different member databases, so their e-values
                and/or scoring systems cannot be meaningfully compared or
                combined. For this reason, we do not show e-values on the
                InterPro web site. However, e-values can be obtained via the
                downloadable InterProScan software package, which outputs
                detailed individual results for each member database sequence
                analysis algorithm.
              </p>
            </details>

            <details className={f('accordion-style')}>
              <summary>Can I trust my sequence search results?</summary>
              <div>
                <p>
                  We make every effort to ensure that signatures integrated into
                  InterPro are accurate. Before being integrated, signatures are
                  manually checked by curators to ensure that they are of a high
                  quality (i.e., they match the proteins they are supposed to
                  and hit as few incorrect proteins as possible).
                </p>
                <p>
                  While matches to InterPro should therefore be trustworthy,
                  there are some caveats. Most proteins are currently
                  uncharacterised, so quality checks can only ever be based on
                  the subset of characterised proteins that match the signature.
                  It is therefore possible that signatures can match false
                  positives that have not been detected.
                </p>
                <p>
                  A useful rule of thumb is that the more signatures within an
                  InterPro entry that match a protein, the more likely it is
                  that the match is correct. Matches within the same hierarchy
                  would also tend to increase confidence, as they all imply
                  membership of a particular group.
                </p>
                <p>
                  Nevertheless, please bear in mind that the member database
                  signatures are computational predictions. If you think one of
                  our signatures matches false positives, please{' '}
                  <Link
                    href="//www.ebi.ac.uk/support/interpro-general-query"
                    className={f('ext')}
                    target="_blank"
                  >
                    contact us
                  </Link>.
                </p>
              </div>
            </details>

            <details className={f('accordion-style')}>
              <summary>How are InterPro entries mapped to GO terms?</summary>
              <p>
                The assignment of GO terms to InterPro entries is performed
                manually, and is an ongoing process (<Link
                  to={{
                    description: {
                      other: ['help', 'documentation/publication'],
                    },
                  }}
                >
                  view related publication
                </Link>).
              </p>
            </details>

            <details className={f('accordion-style')}>
              <summary>How do I download InterPro?</summary>
              <p>
                The database and related software are freely available from the
                ftp site for download and distribution, provided the following
                Copyright line notice is supplied:{' '}
                <i>
                  InterPro - Integrated Resource Of Protein Domains And
                  Functional Sites. Copyright (C) 2001 The InterPro Consortium.
                </i>
              </p>
            </details>

            <details className={f('accordion-style')}>
              <summary>How do I contribute?</summary>
              <p>
                We welcome your contributions. To report errors or problems with
                the database, please{' '}
                <Link
                  href="//www.ebi.ac.uk/support/interpro-general-query"
                  className={f('ext')}
                  target="_blank"
                >
                  get in touch via EBI support
                </Link>.
              </p>
            </details>

            <details className={f('accordion-style')}>
              <summary>How do I collaborate?</summary>
              <p>
                If you are interested in potential collaborations with InterPro,
                please{' '}
                <Link
                  href="//www.ebi.ac.uk/support/interpro-general-query"
                  className={f('ext')}
                  target="_blank"
                >
                  contact us via EBI support
                </Link>.
              </p>
            </details>
          </div>
          <div className={f('columns', 'large-4')}>
            <div className={f('box-add')}>
              <h3 className={f('light', 'margin-top-xxlarge')}>
                Additional help
              </h3>
              <ul>
                {
                  // <li>
                  //   <Link href="//www.ebi.ac.uk/support/interpro-general-query">
                  //     <span
                  //       className={f('icon', 'icon-common')}
                  //       data-icon="&#x27a;"
                  //     />{' '}
                  //     Chat with an assistant{' '}
                  //   </Link>
                  // </li>
                }
                <li>
                  <Link href="//www.ebi.ac.uk/support/interpro-general-query">
                    <span
                      className={f('icon', 'icon-common')}
                      data-icon="&#x1d8;"
                    />{' '}
                    Submit a ticket
                  </Link>
                </li>
              </ul>
            </div>
          </div>
        </div>
      </section>
    );
  }
}
