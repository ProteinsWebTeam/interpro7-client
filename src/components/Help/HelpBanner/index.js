import React from 'react';
import T from 'prop-types';
import { foundationPartial } from 'styles/foundation';

import fonts from 'EBI-Icon-fonts/fonts.css';
import style from './style.css';
import Link from 'components/generic/Link';

const f = foundationPartial(style, fonts);

const helpTopics = {
  InterProScan: {
    documentation: {
      to: { description: { other: ['help', 'documentation'] } },
    },
    training: {
      href:
        'http://www.ebi.ac.uk/training/online/course/interpro-functional-and-structural-analysis-protei/sequence-searching',
      target: '_blank',
    },
  },
  IDA: {
    documentation: {
      to: { description: { other: ['about', 'ida'] } },
    },
  },
  TextSearch: {
    documentation: {
      href: 'http://www.ebi.ac.uk/ebisearch/documentation.ebi',
      target: '_blank',
    },
  },
  default: {
    documentation: {
      to: { description: { other: ['help', 'documentation'] } },
    },
  },
};

export const HelpBanner = ({ topic } /*: {topic: string} */) => {
  const current = helpTopics[topic] || helpTopics.default;
  return (
    <div className={f('help-banner', 'flex-card')}>
      <header>
        <span
          className={f('icon', 'icon-common', 'font-l')}
          data-icon="&#xf059;"
        />{' '}
        Need more Help?
      </header>
      <div>
        If you need more info on {topic}, you can either look at the:
        <br />
        <span
          className={f('icon', 'icon-common', 'font-l')}
          data-icon="&#xf02d;"
        />{' '}
        <Link
          {...current.documentation}
          className={f({ ext: current.documentation.target })}
        >
          Documentation page
        </Link>
        <br />
        {current.training && (
          <>
            <span
              className={f('icon', 'icon-common', 'font-l')}
              data-icon="&#xf2ff;"
            />{' '}
            <Link
              {...current.training}
              className={f({ ext: current.training.target })}
            >
              Online training course
            </Link>
            <br />
          </>
        )}
        or{' '}
        <Link
          href="http://www.ebi.ac.uk/support/interpro-general-query"
          target="_blank"
        >
          contact us
        </Link>{' '}
        directly with your question.
      </div>
    </div>
  );
};
HelpBanner.propTypes = {
  topic: T.string,
};

export default React.memo(HelpBanner);
