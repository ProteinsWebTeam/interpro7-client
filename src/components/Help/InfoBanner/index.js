import React from 'react';
import T from 'prop-types';
import { foundationPartial } from 'styles/foundation';

import style from './style.css';
import cards from 'components/SimpleCommonComponents/Card/styles.css';
import ipro from 'styles/interpro-new.css';

const f = foundationPartial(style, ipro, cards);

const infoTopics = {
  InterProScan: {
    text: 'InterProScan is a sequence analysis application (nucleotide and protein sequences) that combines different protein signature recognition methods into one resource.',
    image: 'image-tool-ipscan',
  },
  IDA: {
    text: 'The InterPro Domain Architecture (IDA) tool allows you to search the InterPro database with a particular set of domains, and returns all of the domain architectures and associated proteins that match the query.',
    image: 'image-tool-ida',
  },
  TextSearch: {
    text: 'InterPro text search is powered by the EBI Search, a scalable text search engine that provides easy and uniform access to the biological data resources hosted at the European Bioinformatics Institute (EMBL-EBI).',
    image: 'image-tool-textsearch',
  },
  default: {},
};

export const InfoBanner = ({ topic } /*: {topic: string} */) => {
  const current = infoTopics[topic] || infoTopics.default;
  return (
    <div className={f('help-banner', 'flex-card')}>
      <div className={f('card-image', current.image)}>
        <div className={f('card-tag', 'tag-tool')}>{topic}</div>
      </div>
      <div className={f('card-content')}>
        <div className={f('card-description')}>{current.text}</div>
      </div>
    </div>
  );
};
InfoBanner.propTypes = {
  topic: T.string,
};

export default React.memo(InfoBanner);
