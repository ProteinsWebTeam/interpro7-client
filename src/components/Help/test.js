import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import Help from '.';

const renderer = new ShallowRenderer();

describe('<Help />', () => {
  test('should render', () => {
    renderer.render(<Help />);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});
