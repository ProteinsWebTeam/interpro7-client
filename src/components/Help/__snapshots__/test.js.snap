// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Help /> should render 1`] = `
<div
  className="help-flex"
>
  <div
    className="vf-grid vf-grid__col-3 help-grid"
  >
    <div
      className="flex-box"
    >
      <span
        className="icon icon-common xl"
        data-icon=""
      />
       
      <h5>
        Getting started
      </h5>
      <li>
        <ul>
          <li>
            <Memo(Connect(Link))
              className="ext"
              href="https://www.ebi.ac.uk/training/online/courses/interpro-quick-tour/"
              target="_blank"
            >
              InterPro quick tour
            </Memo(Connect(Link))>
          </li>
          <li>
            <Memo(Connect(Link))
              className="ext"
              href="https://www.ebi.ac.uk/training/online/courses/pfam-quick-tour/"
              target="_blank"
            >
              Pfam quick tour
            </Memo(Connect(Link))>
          </li>
        </ul>
      </li>
    </div>
    <div
      className="flex-box"
    >
      <span
        className="icon icon-common xl"
        data-icon=""
      />
       
      <h5>
        Training & outreach
      </h5>
      <li>
        <ul>
          <li>
            <Memo(Connect(Link))
              className="ext"
              href="https://www.ebi.ac.uk/training/search-results?query=interpro&domain=ebiweb_training&page=1&facets="
              target="_blank"
            >
              InterPro
            </Memo(Connect(Link))>
          </li>
          <li>
            <Memo(Connect(Link))
              className="ext"
              href="https://www.ebi.ac.uk/training/search-results?query=pfam&domain=ebiweb_training&page=1&facets="
              target="_blank"
            >
              Pfam
            </Memo(Connect(Link))>
          </li>
          <Memo(Connect(Link))
            to={
              {
                "description": {
                  "other": [
                    "help",
                    "protein_families_game",
                  ],
                },
              }
            }
          >
            Protein families game
          </Memo(Connect(Link))>
        </ul>
      </li>
    </div>
    <div
      className="flex-box"
    >
      <span
        className="icon icon-common xl"
        data-icon=""
      />
       
      <h5>
        Documentations
      </h5>
      <li>
        <ul>
          <li>
            <Memo(Connect(Link))
              to={
                {
                  "description": {
                    "other": [
                      "help",
                      "faqs",
                    ],
                  },
                }
              }
            >
              InterPro FAQs
            </Memo(Connect(Link))>
          </li>
          <li>
            <Memo(Connect(Link))
              className="ext"
              href="https://interpro-documentation.readthedocs.io/en/latest/"
              target="_blank"
            >
              InterPro documentation
            </Memo(Connect(Link))>
          </li>
          <li>
            <Memo(Connect(Link))
              className="ext"
              href="https://pfam-docs.readthedocs.io/en/latest/"
              target="_blank"
            >
              Pfam documentation
            </Memo(Connect(Link))>
          </li>
        </ul>
      </li>
    </div>
    <div
      className="flex-box"
    >
      <span
        className="icon icon-common xl"
        data-icon=""
      />
       
      <h5>
        About us
      </h5>
      <li>
        <ul>
          <li>
            <Memo(Connect(Link))
              to={
                {
                  "description": {
                    "other": [
                      "about",
                      "interpro",
                    ],
                  },
                }
              }
            >
              InterPro
            </Memo(Connect(Link))>
          </li>
          <li>
            <Memo(Connect(Link))
              to={
                {
                  "description": {
                    "other": [
                      "about",
                      "consortium",
                    ],
                  },
                }
              }
            >
              InterPro consortium
            </Memo(Connect(Link))>
          </li>
          <li>
            <Memo(Connect(Link))
              to={
                {
                  "description": {
                    "other": [
                      "about",
                      "team",
                    ],
                  },
                }
              }
            >
              Team
            </Memo(Connect(Link))>
          </li>
        </ul>
      </li>
    </div>
    <div
      className="flex-box"
    >
      <span
        className="icon icon-common xl"
        data-icon=""
      />
       
      <h5>
        How to cite us
      </h5>
      <ul>
        <li>
          <Memo(Connect(Link))
            className="ext"
            href="https://interpro-documentation.readthedocs.io/en/latest/citing.html"
            target="_blank"
          >
            Citing InterPro
          </Memo(Connect(Link))>
        </li>
        <li>
          <Memo(Connect(Link))
            className="ext"
            href="https://pfam-docs.readthedocs.io/en/latest/citing-pfam.html"
            target="_blank"
          >
            Citing Pfam
          </Memo(Connect(Link))>
        </li>
      </ul>
    </div>
    <div
      className="flex-box"
    >
      <span
        className="icon icon-common xl"
        data-icon="t"
      />
       
      <h5>
        InterProScan
      </h5>
      <ul>
        <li>
          <Memo(Connect(Link))
            className="ext"
            href="https://interproscan-docs.readthedocs.io/"
            target="_blank"
          >
            InterProScan documentation
          </Memo(Connect(Link))>
        </li>
        <li>
          <Memo(Connect(Link))
            className="ext"
            href="//www.ebi.ac.uk/jdispatcher/docs/"
            target="_blank"
          >
            Job Dispatcher documentation
          </Memo(Connect(Link))>
        </li>
      </ul>
    </div>
    <div
      className="flex-box"
    >
      <span
        className="icon icon-common xl"
        data-icon=""
      />
      <h5>
        Technical corner
      </h5>
      <ul>
        <li>
          <Memo(Connect(Link))
            className="ext"
            href="          https://github.com/ProteinsWebTeam/interpro7-api/tree/master/docs"
            target="_blank"
          >
            Programmatic access
          </Memo(Connect(Link))>
        </li>
        <li>
          <Memo(Connect(Link))
            className="ext"
            href="https://www.ebi.ac.uk/interpro/api/static_files/swagger/"
            target="_blank"
          >
            Swagger UI
          </Memo(Connect(Link))>
        </li>
        <li>
          <Memo(Connect(Link))
            to={
              {
                "description": {
                  "main": {
                    "key": "result",
                  },
                  "result": {
                    "type": "download",
                  },
                },
                "hash": "/entry/interpro/|json",
              }
            }
          >
            Code snippet generator
          </Memo(Connect(Link))>
        </li>
      </ul>
    </div>
  </div>
  <div
    className="box-add"
  >
    <div
      className="vf-box"
    >
      <h3>
         Contact Us 
      </h3>
    </div>
    <div
      className="vf-box"
    >
      <p
        className="help-links"
      >
        InterPro
      </p>
      <ul
        className="vf-list"
      >
        <li>
          <Memo(Connect(Link))
            href="//www.ebi.ac.uk/support/interpro"
            target="_blank"
            withReferrer={true}
          >
            <span
              className="icon icon-common"
              data-icon="n"
            />
             
            Submit a ticket
          </Memo(Connect(Link))>
        </li>
        <li>
          <Twitter />
        </li>
        <li>
          <Memo(Connect(Link))
            href="https://www.linkedin.com/company/interpro-pfam/"
            target="_blank"
            withReferrer={true}
          >
            <span
              className="icon icon-common"
              data-icon=""
            />
             
            LinkedIn account
          </Memo(Connect(Link))>
        </li>
      </ul>
    </div>
    <div
      className="vf-box"
    >
      <p
        className="help-links"
      >
         Pfam 
      </p>
      <ul
        className="vf-list"
      >
        <li>
          <Memo(Connect(Link))
            href="//www.ebi.ac.uk/support/pfam"
            target="_blank"
            withReferrer={true}
          >
            <span
              className="icon icon-common"
              data-icon="n"
            />
             
            Submit a ticket
          </Memo(Connect(Link))>
        </li>
        <li>
          <Twitter
            handler="PfamDB"
          />
        </li>
      </ul>
    </div>
  </div>
</div>
`;
